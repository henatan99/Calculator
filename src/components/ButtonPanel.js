import PropTypes from 'prop-types';
import Button from './Button';

function ButtonPanel(props) {
  const { handleClick } = props;
  return (
    <div className="button-panel">
      <div className="group-1">
        <Button name="AC" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="+/-" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="%" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="รท" handleClick={handleClick} color="#f5913e" wide="25%"/>
      </div>
      <div className="group-2">
        <Button name="7" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="8" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="9" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="X" handleClick={handleClick} color="#f5913e" wide="25%"/>
      </div>
      <div className="group-3">
        <Button name="4" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="5" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="6" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="-" handleClick={handleClick} color="#f5913e" wide="25%"/>
      </div>
      <div className="group-4">
        <Button name="1" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="2" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="3" handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="+" handleClick={handleClick} color="#f5913e" wide="25%"/>
      </div>
      <div className="group-5">
        <Button name="0" handleClick={handleClick} color="#e0e0e0" wide="50%"/>
        <Button name="." handleClick={handleClick} color="#e0e0e0" wide="25%"/>
        <Button name="=" handleClick={handleClick} color="#f5913e" wide="25%"/>
      </div>
    </div>
  );
}

ButtonPanel.propTypes = {
  handleClick: PropTypes.func,
};

ButtonPanel.defaultProps = {
  handleClick: null,
};

export default ButtonPanel;
