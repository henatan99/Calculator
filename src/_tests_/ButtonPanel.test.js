import React from 'react';
import { create } from 'react-test-renderer';
import ButtonPanel from '../components/ButtonPanel';

it('renders when no props passed', () => {
  const tree = create(<ButtonPanel />).toJSON;
  expect(tree).toMatchSnapshot();
});

it('renders when handleClick function prop passed', () => {
  const handleClick = jest.fn();
  const tree = create(<ButtonPanel handleClick={handleClick} />).toJSON;
  expect(tree).toMatchSnapshot();
});
