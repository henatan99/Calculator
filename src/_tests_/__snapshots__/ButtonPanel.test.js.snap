// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when handleClick function prop passed 1`] = `[Function]`;

exports[`renders when no props passed 1`] = `[Function]`;
